### ISSUE TITLE
Data Table Not Accessible to VoiceOver iOS Screen Reader Users

### ISSUE THREAD

**Issue by pauljadam** [vaadin/vaadin-grid#822]
https://vaadin.com/elements/-/element/vaadin-grid#demos

Using the VoiceOver Tables Rotor it says there are 0 tables on this page. 

When you set focus to a cell inside the data table VoiceOver iOS does not speak the caption or the row and column headers for that cell. It only speaks the visible text value unlike a normal accessible data table. 

When you swipe between columns VoiceOver iOS does not speak the column headers like a normal accessible data table. 

Also as a side note there needs to be a way to have row headers <th scope="row"> so the row headers are spoken when the user moves between rows. 

Thanks!

**Comment by Saulis** [comment]
Hi Paul and thanks for the feedback.

Yeah, the current implementation we have in place is considering only keyboard based navigation, unfortunately. 

Our main restriction with the accessibility has been the fact that we are using a virtual DOM which renders and reuses row elements and forced us to omit the built-in table semantics – otherwise screen readers would announce the number of items and current row index incorrectly.

After lots of trial-and-error we ended up using a synthetic focus combined with `aria-labelledby` attributes combined to announce the header and contents for cells.

Obviously not using native focus around the cells while the default table semantics being omitted prevents devices like VoiceOver on iOS from functioning properly. Furthermore, I can't seem to be able to scroll the grid with the three finger gesture on iOS 10 using VoiceOver.

Any fresh ideas and tips how to improve the situation are highly appreciated!

**Comment by pauljadam** [comment]
It's pretty hard to debug the demos to find the exact problem but I can determine it's being caused by the CSS because when you disable styles the table appears to VoiceOver as a real data table and when CSS is on they don't appear as tables at all.

**Comment by Saulis** [comment]
Hmm I wonder if that has something to do with the "outerscroller" element we have on top of the grid which has `pointer-events: none`.

Can you give any tips how I should/could debug this?

**Comment by web-padawan** [comment]
This issue is quite outdated and was submitted for older version of the component.

The a11y support has since been improved so that 4.x and 5.x versions work better.
For more details on the actual screen readers support, see https://github.com/vaadin/vaadin-grid/issues/1015#issuecomment-333854723.
